### ISSUE TITLE
widget relationwith a single collection file

### ISSUE THREAD

**Issue by CarloBu** [sveltia/sveltia-cms#184]
Hello, would it be possible to allow the widget relation to use a single collection file? Right now, it only works with separate files as different relation categories. That’s fine, but in some cases, it’s hard to arrange the order by going into different files and setting the order, and keeping the number order.

`collections:
  - label: Tutorials
    name: tutorials
    media_folder: /tutrialsPreview
    public_folder: /tutrialsPreview
    create: true
    delete: false
    icon: school
    format: json
    extension: json

    files:
      - file: 'src/content/tutorials/tutorials.json'
        label: Tutorial List
        name: tutorialList
        widget: object
        fields:
          - label: Tutorial
            name: tutorial
            widget: list
            fields:
              - {label: Video, name: src, widget: file}
              - {label: Category, dropdown_threshold: 10, name: category, widget: relation, collection: categories, multiple: true, value_field: category}      

  - label: Categories
    name: categories
    create: true
    icon: dashboard_customize
    format: json
    extension: json
    files:
      - file: 'src/content/categories/categories.json'
        label: Categories List
        name: categoriesList
        widget: object
        fields:
          - name: categories
            label: Categories
            widget: list
            fields:
              - name: category
                label: Category
                widget: string
              - name: order
                label: Order
                widget: number`

the current solution is not ideal to arrange the categories order.
![image](https://github.com/user-attachments/assets/e3496b90-d92b-44c8-84f7-94838280ce91)

**Comment by CarloBu** [comment]
summary helps a little bit :) I keep forgetting it.
![image](https://github.com/user-attachments/assets/397d3702-ba0b-4b65-af94-9352da37a1ff)

**Comment by kyoshino** [comment]
Maybe it’s possible with “Referencing a file collection list field example” mentioned in the [Relation widget doc](https://decapcms.org/docs/widgets/#relation)?

**Comment by CarloBu** [comment]
Oh, I didn't catch that. I will try this way, thank you!

**Comment by CarloBu** [comment]
@kyoshino 
I have another question, is there some way to preview videos in the sveltia cms? at least to have a first-frame preview? I thought about creating preview_template for this, but I guess the sveltia does not support it yet.
![image](https://github.com/user-attachments/assets/9a21da03-d151-470a-8b14-30c65e00d540)

**Comment by kyoshino** [comment]
Hmm, video preview should work with the File widget. Will check.

**Comment by CarloBu** [comment]
this is my setup:
`  - label: Tutorials
    name: tutorials
    create: true
    delete: false
    icon: school
    format: json
    extension: json
    files:
      - file: 'src/content/tutorials/tutorials.json'
        label: Tutorial List
        name: tutorialList
        widget: object
        fields:
          - label: Tutorial
            name: tutorial
            widget: list
            fields:
              - {label: Video, name: src, widget: file}
              - {label: Category, dropdown_threshold: 10, name: category, widget: relation, collection: categories, multiple: true, value_field: category}     `

**Comment by CarloBu** [comment]
expected to see like this:
![image](https://github.com/user-attachments/assets/a634c734-fcbd-48fe-9398-955d2d0bd424)

**Comment by kyoshino** [comment]
Nah, video preview is not implemented. I’m working on it.

**Comment by kyoshino** [comment]
@CarloBu btw, did the file collection reference method (`file` option`) work for you? If it doesn’t seem to work as expected, I will take a look.

**Comment by CarloBu** [comment]
Hey, kyoshino,
sorry for the late reply, I can not lie, I was struggling to make a file collection list field work. 
After some painful time, I realized that specifically in the 'category' field of the tutorialList collection, the 'file: categoryListFile' doesn't actually refer to a file name. Instead, it's referencing the name of the file object within the categoryCollection. It's hella counterintuitive at least for me :)

the setup is now working:
```
  - name: tutorialList
    label: Tutorial List
    create: true
    delete: false
    icon: school
    format: json
    extension: json
    files:
      - file: 'src/content/tutorials/tutorials.json'
        label: Tutorial List
        name: tutorialListFile
        widget: object
        fields:
          - label: Tutorial Items
            name: tutorialItems
            widget: list
            fields:
              - {label: Video, name: src, widget: file}
              - label: Category
                name: category
                dropdown_threshold: 10
                widget: relation
                collection: categoryCollection
                file: categoryListFile
                value_field: "categories.*.category"
                multiple: true

  - name: categoryCollection
    label: Categories
    create: true
    delete: true
    files:
      - file: 'src/content/categories/categories.json'
        label: Category List
        name: categoryListFile
        widget: object
        fields:
          - label: Categories
            name: categories
            widget: list
            fields:
              - {label: Category, name: category, widget: string}
```

**Comment by kyoshino** [comment]
No worries, I’ll have a look.

**Comment by kyoshino** [comment]
Just quickly tested your config.

```
file: categoryListFile      <---- why the hell does it not refer to the actual filename :)
```

I think this is how it works 😉 Sveltia CMS could provide a better document for this. I understand that a only clue is the Decap example saying `file: "cities"`, not `file: "/path/to/cities.json"`, which could easily be missed.

```
value_field: "categories.*.category"   <---- and I had to remove brackets for some reason to make it work
```

It worked for me with or without single or double quotations. Not sure why it doesn’t work. It shouldn’t affect the functionality because the YAML parser can handle the differences, and it would be the same string internally.

Anyway, good to know it’s working. The video preview issue has been fixed in [v0.36.4](https://github.com/sveltia/sveltia-cms/releases/tag/v0.36.4), and I’ll close the issue.

**Comment by CarloBu** [comment]
> I think this is how it works 😉 Sveltia CMS could provide a better document for this. I understand that a only clue is the Decap example saying file: "cities", not file: "/path/to/cities.json", which could easily be missed.

I somehow misread this as actual filename, but you right, when read closely it kinda make sense :)

> It worked for me with or without single or double quotations. Not sure why it doesn’t work. It shouldn’t affect the functionality because the YAML parser can handle the differences, and it would be the same string internally.

I don't know why it does not work with brakets for me, maybe my project environment issue

> Anyway, good to know it’s working. The video preview issue has been fixed in [v0.36.4](https://github.com/sveltia/sveltia-cms/releases/tag/v0.36.4), and I’ll close the issue.

awesome! now it's very easy to manage mp4 data, thank you!

**Comment by CarloBu** [comment]
Hey Kyoshino,
I noticed that some mp4 file thumbnails are not showing, like this one:
![image](https://github.com/user-attachments/assets/e340d263-1116-49f0-b9b9-26bad4551b50)

https://github.com/user-attachments/assets/00030e52-4bef-4e0d-b1f3-a65a7e6b35c5

But it's totally fine for me, the most important thing is that I can preview and watch all mp4 videos 👍 I just wanted to give awareness about this bug

**Comment by kyoshino** [comment]
Maybe some videos take a bit longer to show the first frame. I will find a way to generate thumbnails more reliably.

**Comment by CarloBu** [comment]
maybe, but only specific video thumbnails are not showing. Later I will check deeper if these video thumbnails show up in other setup and test projects. Anyways this is a low-priority issue.

**Comment by kyoshino** [comment]
I’ve just shipped [v0.36.5](https://github.com/sveltia/sveltia-cms/releases/tag/v0.36.5) to solve the video thumbnail issue. You need to open DevTools and clear the `asset-thumbnails` cache in IndexedDB to regenerate the thumbnails.

<img width="628" alt="image" src="https://github.com/user-attachments/assets/215b3251-9b52-4955-9d14-7c849cf398e1">

**Comment by CarloBu** [comment]
Ooowee! It's now working perfectly! Thank you!
![image](https://github.com/user-attachments/assets/7eeecc14-70e2-4fa7-a2cd-f96e00d5ceba)
I see the issue was with the timing, you added 250ms timer to wait for the first frame to render, smart 👍
